dotCMS/src/main/java/com/dotmarketing/util/starter/ExportStarterUtil.java

```diff
@@ -405,4 +721,110 @@ public Optional<File> zipStarter(final OutputStream output, final boolean withAs
 
         return Optional.of(newLocation);
     }
+
+    /**
+     * Generates all the contents that go into the dotCMS Starter zip file. It's very important to point out that <b>no
+     * temporary files are created, nor any piece of data is written to the FS at any point.</b>
+     * <p>The resulting ZIP file is fully created in memory and, as soon as the first entry is added, it is directly
+     * streamed to the {@link OutputStream} for users to be able to download it even before it is 100% ready.</p>
+     *
+     * @param output The {@link OutputStream} instance.
+     * @param includeAssets If the Starter file must contain all dotCMS assets as well, set this to {@code true}.
+     */
+    public void streamZipStarter(final OutputStream output, final boolean includeAssets) {
+        final List<FileEntry> starterData = this.getStarterDataAsJSON();
+        if (includeAssets) {
+            starterData.addAll(this.getAssets());
+        }
+        this.createInMemoryZip(starterData, output);
+    }
+
+    /**
+     * Creates an in-memory ZIP file with the provided list of contents. The output object provided by the
+     * {@link StreamingOutput} is passed down to the ZIP generation process so that the actual file download as soon as
+     * the first entry is added to it.
+     *
+     * @param output      The {@link OutputStream} instance that will be sent back in the response.
+     * @param starterData The complete list of dotCMS data structures -- and optionally assets -- that must be added
+     *                    to the in-memory ZIP file.
+     */
+    private void createInMemoryZip(final List<FileEntry> starterData, final OutputStream output) {
```

Comment:
Maybe rename this to `createStreamingZip`. The name scared be because in many cases the zip we are creating is going to be many many GB in size, enough to blow any java Heap.

Reply:
Right, that name is way better. Done!